### PR TITLE
Patched packages grouping

### PR DESCRIPTION
**Pull request by CraigRichards**
Hi Nick,
I found an issue with Package Grouping and the generated report for this project shows the updates.

Do you have plans to publish your refactoring branch soon or should I have made the changes elsewhere? It looked like a good starting point and I like the project refactor to get the tests running.

Cheers
Craig

### CODE REVIEW

File: src/allure-nunit-tests/allure-results/environment.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<environment>
```
Comment (unickq) [inline src/allure-nunit-tests/allure-results/environment.xml:1]: why this file is here?

File: src/allure-nunit-tests/allure-results/executor.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (unickq) [inline src/allure-nunit-tests/allure-results/executor.json:1]: Same question

File: src/allure-nunit-tests/allureConfig.json [hunk 3]
```diff
@@ -1,13 +1,41 @@
...
-	}
+  "allure": {
+    "directory": "allure-results",
+    "links": [
+      "{link}",
+      "{issue}",
+      "{tms}"
+    ]
+  },
+  "specflow": {
```
Comment (unickq) [inline src/allure-nunit-tests/allureConfig.json:10]: There's no Specflow in this project, so why did you add this?

**Comment by unickq** [comment]
Hi @CraigRichards 

I plan to merge the refactoring branch once I can confirm nothing is broken with it. Just have no time for it now

File: src/allure-nunit-tests/allure-results/environment.xml (hunk 1 above)
Comment (CraigRichards) [reply src/allure-nunit-tests/allure-results/environment.xml:1]: I added sample configuration files for this setting so I could see it flow through to the local report. 

Might be a useful reference for anyone looking into producing a fully featured report. 

Happy to amend and remove.

File: src/allure-nunit-tests/allure-results/executor.json (hunk 2 above)
Comment (CraigRichards) [reply src/allure-nunit-tests/allure-results/executor.json:1]: I added sample configuration files for this setting so I could see it flow through to the local report. 

Might be a useful reference for anyone looking into producing a fully featured report. 

Happy to amend and remove.

File: src/allure-nunit-tests/allureConfig.json (hunk 3 above)
Comment (CraigRichards) [reply src/allure-nunit-tests/allureConfig.json:10]: Quick question,
I noticed you extended AllureConfiguration to include some customisations. I need to do the same thing myself.

I think these customizations should live in an nunit configuration section like SpecFlow as they only affect the generation of NUnit tests.

Backward compatability is an issue and a fallback to the old location is an option.

What do you think?

File: src/allure-nunit-tests/allure-results/executor.json (hunk 2 above)
Comment (unickq) [reply src/allure-nunit-tests/allure-results/executor.json:1]: I think it's better to remove it

File: src/allure-nunit-tests/allure-results/environment.xml (hunk 1 above)
Comment (unickq) [reply src/allure-nunit-tests/allure-results/environment.xml:1]: Ok, if it shown here - let's keep

image

File: src/allure-nunit/Core/AllureNUnitHelper.cs [hunk 4]
```diff
@@ -65,9 +65,9 @@ private void StartTestCase()
                 {
                     Label.Thread(),
                     Label.Host(),
-                    Label.Package(_test.ClassName?.Substring(0, _test.ClassName.LastIndexOf('.'))),
+                    Label.Package(_test.FullName),
                     Label.TestMethod(_test.MethodName),
-                    Label.TestClass(_test.ClassName?.Substring(_test.ClassName.LastIndexOf('.') + 1))
```
Comment (unickq) [inline src/allure-nunit/Core/AllureNUnitHelper.cs:70]: I made this on purpose TestClass - test class name.

File: src/allure-nunit/Core/AllureNUnitHelper.cs [hunk 5]
```diff
@@ -65,9 +65,9 @@ private void StartTestCase()
                 {
                     Label.Thread(),
                     Label.Host(),
-                    Label.Package(_test.ClassName?.Substring(0, _test.ClassName.LastIndexOf('.'))),
```
Comment (unickq) [inline src/allure-nunit/Core/AllureNUnitHelper.cs:68]: I made this on-purpose Package - test namespace.

File: src/allure-nunit-tests/allureConfig.json (hunk 3 above)
Comment (unickq) [reply src/allure-nunit-tests/allureConfig.json:10]: I think it can go to docs, not. Please keep original allureConfig.json except `brokenTestData`

File: src/allure-nunit-tests/allure-results/launch.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "name": "Demo Allure tests"
```
Comment (unickq) [inline src/allure-nunit-tests/allure-results/launch.json:2]: Didn't find where it goes with allure report UI - don't see the reason too keep this file

File: src/allure-nunit-tests/allure-results/executor.json (hunk 2 above)
Comment (unickq) [reply src/allure-nunit-tests/allure-results/executor.json:1]: This file is usually generated by CI plugins. Same as tests trend, job history, etc.
I'd add this to docs later

**Review by unickq — CHANGES_REQUESTED** [review]
Finally had some time to open your PR :)
Please see my comments

File: src/allure-nunit-tests/allure-results/executor.json (hunk 2 above)
Comment (CraigRichards) [reply src/allure-nunit-tests/allure-results/executor.json:1]: > This file is usually generated by CI plugins. Same as tests trend, job history, etc.
> I'd add this to docs later

This is good to know. I generate all my reports manually, copy history etc.

Have you integrated into TeamCity / Azure DevOps using this approach?
I'm very interested as I love what Allure can do.

**Comment by CraigRichards** [comment]
> Hi @CraigRichards
> 
> I plan to merge the refactoring branch once I can confirm nothing is broken with it. Just have no time for it now

I understand, take your time mate and look forward to contributing my ideas to this ecosystem.  I've been using your Nunit package for almost 3.5 years.

**Comment by unickq** [comment]
> I understand, take your time mate and look forward to contributing my ideas to this ecosystem

Yep, I reviewed your changes and left some comments

**Comment by unickq** [comment]
Any updates on that?

**Comment by unickq** [comment]
No activity for a year
